client/my-sites/patterns/components/page-view-tracker.tsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect } from 'react';
+import PageViewTracker from 'calypso/lib/analytics/page-view-tracker';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { useSelector } from 'calypso/state';
+import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
+import getUserSetting from 'calypso/state/selectors/get-user-setting';
+
+type PatternsPageViewTrackerProps = {
+	category: string;
+	searchTerm: string;
+};
+
+export function PatternsPageViewTracker( { category, searchTerm }: PatternsPageViewTrackerProps ) {
+	const isLoggedIn = useSelector( isUserLoggedIn );
+	const isDevAccount = useSelector( ( state ) => getUserSetting( state, 'is_dev_account' ) );
+
+	useEffect( () => {
+		const commonProperties = {
+			is_logged_in: isLoggedIn,
+			user_is_dev_account: isDevAccount ? '1' : '0',
+		};
+
+		if ( searchTerm ) {
+			recordTracksEvent( 'calypso_pattern_library_search', {
+				...commonProperties,
+				search_term: searchTerm,
+			} );
+		}
+
+		if ( category ) {
+			recordTracksEvent( 'calypso_pattern_library_filter', {
+				...commonProperties,
+				category,
+			} );
+		}
+	}, [ category, isDevAccount, isLoggedIn, searchTerm ] );
```

Comment:
It strikes me that even with the `else if`, we we'd could still end up reporting duplicate events, because `useEffect()` is triggered whenever any one of `searchTerm` and `category` is updated. Imagine the scenario where:
* User selects a category -> category filter event is tracked as expected
* User performs a search -> search event is tracked as expected
* User selects another category -> both category and filter events are tracked, when we only expect the category filter event to be tracked. We ended up two events because the change of `category` triggered the `useEffect` hook and the emptying of the search term triggered the `useEffect` hook a second time.

I attempted to fix the issue with the following change:

```suggestion
	const [ currentQuery, setCurrentQuery ] = useState( { category, searchTerm } );

	useEffect( () => {
		const commonProperties = {
			is_logged_in: isLoggedIn,
			user_is_dev_account: isDevAccount ? '1' : '0',
		};

		if ( category !== currentQuery.category ) {
			recordTracksEvent( 'calypso_pattern_library_filter', {
				...commonProperties,
				category,
			} );

		} else if ( searchTerm && currentQuery.searchTerm !== searchTerm ) {
			recordTracksEvent( 'calypso_pattern_library_search', {
				...commonProperties,
				search_term: searchTerm,
			} );
		}

		setCurrentQuery( { category, searchTerm } );

	}, [ category, isDevAccount, isLoggedIn, searchTerm ] );
```

While the above change managed to ensure we only track one of `calypso_pattern_library_filter` or `calypso_pattern_library_search`, we still end up with two `calypso_page_view` events, one tracking a path without `:searchTerm`, another attributing a path with `:searchTerm`. I'll leave it for you to continue the investigation on how to go about solving that issue.